### PR TITLE
add delete account functionality for user

### LINKED ISSUE #1421: Add option for users to delete their account
As a user
So that I can not use or be seen on the site
I want to be able to delete/deactivate my account

### PR DESCRIPTION
**Pull request by mattlindsey**
#### Description
This change adds and enables a button on the user's profile to delete their account, which actually does a 'soft delete' due to a previous PR.

#### Motivation and Context
fixes #1421 

#### How Has This Been Tested?
Locally I deleted several accounts and saw that they disappeared from the website, but the record still existed with deleted_at set

#### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

#### Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: features/step_definitions/user_steps.rb [hunk 1]
```diff
@@ -402,3 +406,7 @@
   link = user_path(user)
   page.find(:css, %Q{a[href="#{link}?tab=activity"]}).trigger('click')
 end
+
+Then(/^I should not exist as a user$/) do
+  expect(@user).to be_nil
```
Comment (tansaku) [inline features/step_definitions/user_steps.rb]: I think we should reload user from db here to avoid possibility that we are dealing with cached data

**Comment by tansaku** [comment]
@diraulo @mattlindsey tested this locally and it all worked  - made a minor comment, but this is basically ready to go - argument could be made from a UX POV to put deactivate the account from within the edit view since we want to avoid people deactivating accidentally ...

File: features/step_definitions/user_steps.rb (hunk 1 above)
Comment (mattlindsey) [inline features/step_definitions/user_steps.rb]: hi @tansaku.  I think this is ok.  and I'm not sure how to reload the user given that @user has been destroyed by the delete_user method.

Comment (tansaku) [inline features/step_definitions/user_steps.rb]: @mattlindsey there is a with_deleted method that allows us to reload it - see the README for the gem

**Comment by tansaku** [comment]
@mattlindsey - biggest priority here I would say is moving the functionality into the edit screen - can you do that?

File: features/step_definitions/user_steps.rb (hunk 1 above)
Comment (mtc2013) [inline features/step_definitions/user_steps.rb]: i think, offhand, it's a scope attached to the model like -- User.with_deleted.find_by(name: "blah") @mattlindsey @tansaku

**Pull request closed without merging**